pkgs/development/compilers/chicken/5/eggDerivation.nix

```diff
@@ -17,14 +17,16 @@ in
 (stdenv.mkDerivation ({
   name = "chicken-${name}";
   propagatedBuildInputs = buildInputs;
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ chicken makeWrapper ];
   buildInputs = [ chicken ];
 
+  strictDeps = true;
+
   CSC_OPTIONS = lib.concatStringsSep " " cscOptions;
 
   buildPhase = ''
     runHook preBuild
-    chicken-install -cached -no-install ${lib.escapeShellArgs chickenInstallFlags}
+    chicken-install -cached -no-install -host ${lib.escapeShellArgs chickenInstallFlags}
```

Comment:
What is chicken's cross terminology? I'm a bit confused that we pass `-host` to match `hostPlatform`, but use `stdenv.cc` for `TARGET_C_COMPILER` which will produce binaries for `hostPlatform`, not `targetPlatform`.

What does `-host` mean for chicken and what does the lack thereof mean? Does it use autoconf terminology?

Reply:
Some reference documentation:

https://wiki.call-cc.org/man/5/Cross%20development

https://manpages.debian.org/testing/chicken-bin/chicken-install.1.en.html

Unlike some simpler programs, chicken can be made aware of cross compilation and therefore introduces some terminology for it. This gets a bit confusing when combined with nix's cross compilation capabilities.

>  but use stdenv.cc ... which will produce binaries for hostPlatform, not targetPlatform

I think this is wrong. Keep in mind, that we allow cross compilation with the package prefix `pkgsCross` and this sets the `$CC` to the target compiler. The setting of `TARGET_C_COMPILER` in the chicken derivation is that the binaries can be made aware of the full compiler path, which eliminates the need of the wrapper, which can come in handy in a later PR.

So when `chicken-install` normally runs it will try to compile the egg for both, what it thinks is the host and the target. This does not play correctly with the stdenv that is being set and therefore we only build the "host" variant, which is in fact the target variant. You can check this by looking at the architecture of a cross built egg. Also without setting this flag, the build fails.

I found this only to be an implementation detail but if you want I can add some comments about this in an upcoming PR.